packages/api/src/controllers/asset.ts

```diff
@@ -1027,6 +1030,15 @@ app.patch(
       }
     }
 
+    if (
+      (playbackPolicy && asset.playbackPolicy?.type === "webhook") ||
+      playbackPolicy?.type === "webhook"
```

Comment:
Perhaps we actually need a helper like `isPrivatePlaybackPolicy` which does the `!== public` check, and use it here and in the other place where we choose the OS ID.

Here, the logic we should check is
```suggestion
      isPrivatePlaybackPolicy(playbackPolicy) !== isPrivatePlaybackPolicy(asset.playbackPolicy)
```
WDYT?

Reply:
Makes sense, I arrived late on this and now I don't want to touch too much, can open a separate PR

Reply:
Yeah do it. This is not only a nit, the functionality is incomplete (its ignoring jwt policies for example)